octoprint_findmymrbeam/__init__.py

```diff
@@ -61,12 +60,45 @@ def __init__(self):
 		self._all_secrets = self._not_so_secret + [self._secret]
 
 	def initialize(self):
+		self._uuid = self._get_setting([["plugins", "discovery", "upnpUuid"], ], ["public", "uuid"])
+		if self._uuid is None:
+			self._generate_uuid()
+		self._search_id = self._settings.get(["public", "search_id"])
+		if self._search_id is None:
+			self._generate_search_id()
 		self._analytics = Analytics(self)
 		self._url = self._settings.get(["url"])
 		self._logger.info("FindMyMrBeam enabled: %s", self.is_enabled())
 		self._analytics.log_enabled(self.is_enabled())
 		self.update_frontend()
 
+	##~~ data providers ##
+	def _get_server_registry_data(self):
+		return dict(_version=__version__,
+					uuid=self._uuid,
+					search_id=self._search_id,
+					name=self._find_name(),
+					hostname=socket.gethostname(),
+					local_ips=self._get_local_ips(),
+					netconnectd_state=self._get_netconnectd_state(),
+					modes=self._get_internal_modes(),
+					query="plugin/{}/{}".format(self._identifier, self._secret),
+					plugin_version=self._get_plugin_version(),
+					)
+
+	def _get_local_ping_data(self):
+		return dict(_version=__version__,
+					uuid=self._uuid,
+					search_id=self._search_id,
+					name=self._find_name(),
+					hostname=socket.gethostname(),
+					local_ips=self._get_local_ips(),
+					netconnectd_state=self._get_netconnectd_state(),
+					modes=self._get_internal_modes(),
+					query="plugin/{}/{}".format(self._identifier, self._secret),
+					plugin_version=self._get_plugin_version(),
+					)
```

Comment:
No difference between the functions? If there is a difference, please make it more obvious, like making a separate function that contains all the keys except the few keys these functions modify.